azext_iot/digitaltwins/command_map.py

```diff
@@ -45,6 +45,31 @@ def load_digitaltwins_commands(self, _):
         cmd_group.wait_command("wait", "wait_instance")
         cmd_group.command("reset", "reset_instance", confirmation=True, is_preview=True)
 
+    with self.command_group(
+        "dt data-history",
+        command_type=digitaltwins_resource_ops,
+        is_preview=True
+    ) as cmd_group:
+        pass
+
+    with self.command_group(
+        "dt data-history connection",
+        command_type=digitaltwins_resource_ops,
+        is_preview=True
+    ) as cmd_group:
+        cmd_group.show_command("show", "show_adx_data_connection")
+        cmd_group.wait_command("wait", "wait_adx_data_connection")
+        cmd_group.command("list", "list_adx_data_connection")
+        cmd_group.command(
+            "delete", "delete_adx_data_connection", confirmation=True, supports_no_wait=True
+        )
```

Comment:
small nit - if other types of data connections are added, we may want to move these into a `show_data_connection` method, and with `list` as an example - maybe add a `--type adx` param as a possible filter

Reply:
Everything but 'create [connection type]', is meant to generically support data history connections, so agreed the method callback names should not be specific for adx.

Filter wise the built in jmespath --query support _should_ work out of the box (reducing criticality of an explicit filter param/arg) but the net benefit is low because today one kind of connection (adx) can be made.

Reply:
Also there is a general theme of one connection per digital twin so far. I have list in case this changes in the future.

As paymaun said, there is little need for adding a type filter right now.